misk-redis/src/test/kotlin/misk/redis/AbstractRedisTest.kt

```diff
@@ -955,4 +960,1592 @@ abstract class AbstractRedisTest {
 
     assertNull(redis.zscore("foo", "s"));
   }
+
+  @Test fun `zrange by index test - happy case`() {
+    val key = "bar1"
+
+    redis.zadd(
+      key,
+      mapOf(
+        "b" to 5.0,
+        "bz" to 2.3,
+        "bb" to 4.5,
+        "yy" to 6.7,
+        "ad" to 10.0
+      )
+    )
+    // standard happy case. get first four lowest score members.
+    assertEquals(
+      listOf(
+        "bz".encodeUtf8(),
+        "bb".encodeUtf8(),
+        "b".encodeUtf8(),
+        "yy".encodeUtf8()
+      ),
+      redis.zrange(
+        key,
+        INDEX,
+        ZRangeIndexMarker(0),
+        ZRangeIndexMarker(3)
+      )
+    )
+    assertEquals(
+      listOf(
+        Pair(
+          "bz".encodeUtf8(),
+          2.3
+        ),
+        Pair(
+          "bb".encodeUtf8(),
+          4.5
+        ),
+        Pair(
+          "b".encodeUtf8(),
+          5.0
+        ),
+        Pair(
+          "yy".encodeUtf8(),
+          6.7
+        )
+      ),
+      redis.zrangeWithScores(
+        key,
+        INDEX,
+        ZRangeIndexMarker(0),
+        ZRangeIndexMarker(3)
+      )
+    )
+    assertEquals(
+      listOf(
+        "ad".encodeUtf8(),
+        "yy".encodeUtf8(),
+        "b".encodeUtf8(),
+        "bb".encodeUtf8()
+      ),
+      redis.zrange(
+        key,
+        INDEX,
+        ZRangeIndexMarker(0),
+        ZRangeIndexMarker(3),
+        true
+      )
+    )
+    assertEquals(
+      listOf(
+        Pair(
+          "ad".encodeUtf8(),
+          10.0
+        ),
+        Pair(
+          "yy".encodeUtf8(),
+          6.7
+        ),
+        Pair(
+          "b".encodeUtf8(),
+          5.0
+        ),
+        Pair(
+          "bb".encodeUtf8(),
+          4.5
+        )
+      ),
+      redis.zrangeWithScores(
+        key,
+        INDEX,
+        ZRangeIndexMarker(0),
+        ZRangeIndexMarker(3),
+        true
+      )
+    )
+  }
+
+  @Test fun `zrange by index test - stop index exceeds the size`() {
+    val key = "bar1"
+
+    redis.zadd(
+      key,
+      mapOf(
+        "b" to 5.0,
```

Comment:
Are all of these test fixtures different? If not, can the be collected to reduce the LOC size of each test case for better readability/maintainability?

If each text fixture set is unique then it's fine as is.

Reply:
it is same in many but not all places.

Reply:
Would it be worth DRY-ing it up? I worry about maintenance of test files like this with unnecessary duplication and verbosity. Readability is also impacted vs. pulling most of the test fixtures into common variables so it's clear what is new in each test case.